src/feature_extraction/feature_extractor.rs

```diff
@@ -1,18 +1,28 @@
 use pyo3::prelude::*;
+use polars_core::prelude::*;
+//use polars_lazy::prelude::*;
+use polars_core::prelude::*;
+//use polars_lazy::prelude::*;
 use pyo3_polars::{derive::polars_expr, export::polars_core::{series::Series, prelude::{*}}};
 use std::collections::HashSet;
 
-#[pyfunction]
-pub fn rs_lempel_ziv_complexity(
-    s: &[u8]
-) -> usize {
+
+#[polars_expr(output_type=UInt32)]
+fn pl_lempel_ziv_complexity(inputs: &[Series]) -> PolarsResult<Series>  {
+    
+    let input: &Series = &inputs[0];
+    let name = input.name();
+    let input = input.bool()?;
+    let bits: Vec<bool> = input.into_iter().map(
+        |op_b| op_b.unwrap_or(false)
+    ).collect();
 
     let mut ind:usize = 0;
     let mut inc:usize = 1;
 
-    let mut sub_strings: HashSet<&[u8]> = HashSet::new();
-    while ind + inc <= s.len() {
-        let subseq: &[u8] = &s[ind..ind+inc];
+    let mut sub_strings: HashSet<&[bool]> = HashSet::new();
```

Comment:
@abstractqqq rusts default hastable/hashset has a very expensive hash function.

I recommended using hashbrown here.

Reply:
Resolved in #113